upload/admin/model/user/user_group.php

```diff
@@ -75,10 +75,12 @@ public function addPermission($user_group_id, $type, $route) {
 	public function removePermission($user_group_id, $type, $route) {
 		$user_group_query = $this->db->query("SELECT DISTINCT * FROM `" . DB_PREFIX . "user_group` WHERE `user_group_id` = '" . (int)$user_group_id . "'");
 
-		if ($user_group_query->num_rows) {
+		if ($user_group_query->num_rows && $user_group_query->row['permission']) {
```

Comment:
The $user_group_query->row['permission'] field is a JSON string. There is no point on validating that with a simple IF statement. By adding a validation on the type field, this would also prevent error triggers to be outputted as to provide limited support originating from the installed extensions. I would not suggest this.